packages/typescript-to-proptypes/src/injector.ts

```diff
@@ -97,6 +98,13 @@ function getUsedProps(
             babelTypes.isObjectPattern(path.node.id)
           ) {
             getUsedPropsInternal(path.node.id);
+          } else if (
+            // currently tracking `inProps` which stands for the given props e.g. `function Modal(inProps) {}`
+            babelTypes.isIdentifier(node, { name: 'inProps' }) &&
+            // `const props = ...` assuming the right-hand side has `inProps` as input.
+            babelTypes.isIdentifier(path.node.id, { name: 'props' })
+          ) {
+            getUsedPropsInternal(path.node.id);
```

Comment:
It's not pretty since this function has the same issue as most of the code inlined from ttp: Overloading functions to increase re-usability instead of splitting them for individual use cases. 

I expect that this kind of props tracking through throughout the component might increase with the removal of `withStyles` at which point we can refactor this code.